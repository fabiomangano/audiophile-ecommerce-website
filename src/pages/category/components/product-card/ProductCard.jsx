import React from 'react';
import PropTypes from 'prop-types';
import { useMediaQuery } from 'react-responsive';
import { LargeScreen } from '../../../../mediaQueries';
import Marginer from '../../../../components/marginer/Marginer';
import Button from '../../../../components/button/Button';
import style from './ProductCard.module.scss';

const ProductCard = ({ name, description, urls, onClick, reverse }) => {
  const isLargeScreen = useMediaQuery(LargeScreen);
  return (
    <div
      className={
        reverse && isLargeScreen ? style.container_reverse : style.container
      }
    >
      <picture>
        <source srcSet={urls.desktop} media="(min-width:1024px)" />
        <source srcSet={urls.tablet} media="(min-width:768px)" />
        <img src={urls.mobile} alt="New Product" className={style.image} />
      </picture>
      <Marginer direction="horizontal" margin={2} />
      <div className={style.content}>
        <h2 className={style.title}>{name}</h2>
        <Marginer direction="horizontal" margin={2} />
        <p className={style.text}>{description}</p>
        <Marginer direction="horizontal" margin={2} />
        <Button variant="primary" onClick={onClick}>
          SEE PRODUCT
        </Button>
      </div>
    </div>
  );
};

ProductCard.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  urls: PropTypes.arrayOf(String).isRequired,
  onClick: PropTypes.func.isRequired,
  reverse: PropTypes.bool.isRequired,
};

export default ProductCard;
