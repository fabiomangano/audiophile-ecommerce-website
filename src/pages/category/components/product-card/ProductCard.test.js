import React from 'react';
import { create } from 'react-test-renderer';
import ProductCard from './ProductCard';

describe('ProductCard', () => {
  let tree;
  let onClick;

  beforeEach(() => {
    onClick = jest.fn();

    tree = create(
      <ProductCard
        name="product"
        description="some-description"
        url="some-image-url"
        onClick={onClick}
      />,
    );
  });

  it('should render correctly and display name, description, imageUrl of the given product', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('should call onClick when user click on the page', () => {
    const instance = tree.root;
    const button = instance.findByType('button');
    button.props.onClick();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
