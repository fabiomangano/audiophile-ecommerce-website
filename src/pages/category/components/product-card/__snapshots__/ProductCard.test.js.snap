// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render correctly and display name, description, imageUrl of the given product 1`] = `
<div
  className="container"
>
  <img
    alt="Product"
    className="image"
    src="some-image-url"
  />
  <span
    style={
      Object {
        "display": "flex",
        "height": "2rem",
      }
    }
  />
  <h2
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    product
  </h2>
  <span
    style={
      Object {
        "display": "flex",
        "height": "2rem",
      }
    }
  />
  <p
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    some-description
  </p>
  <span
    style={
      Object {
        "display": "flex",
        "height": "2rem",
      }
    }
  />
  <button
    className="primary md"
    onClick={[MockFunction]}
    size="md"
    type="button"
    variant="primary"
  >
    SEE PRODUCT
  </button>
</div>
`;
